### PR TITLE
Add fail on active sync propety

### PR DESCRIPTION
**Pull request by aballiet**
Fixes https://github.com/kestra-io/plugin-airbyte/issues/44

### CODE REVIEW

File: src/main/java/io/kestra/plugin/airbyte/AbstractAirbyteConnection.java [hunk 1]
```diff
@@ -87,6 +85,12 @@ protected <REQ, RES> HttpResponse<RES> request(RunContext runContext, MutableHtt
                 return client.toBlocking().exchange(request, argument);
             }
         } catch (HttpClientResponseException e) {
+            if (e.getStatus().getCode() == 409 && e.getResponse().getBody(String.class).isPresent()){
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/airbyte/AbstractAirbyteConnection.java:88]: Can really a 409 be thrown for something else?

File: src/main/java/io/kestra/plugin/airbyte/connections/Sync.java [hunk 2]
```diff
@@ -75,23 +75,42 @@ public class Sync extends AbstractAirbyteConnection implements RunnableTask<Sync
     @Builder.Default
     Duration pollFrequency = Duration.ofSeconds(1);
 
+    @Schema(
+            title = "Specify whether task should fail if sync is already running"
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/airbyte/connections/Sync.java:79]:
```suggestion
            title = "Specify whether task should fail if a sync is already running"
```

File: src/main/java/io/kestra/plugin/airbyte/connections/Sync.java [hunk 3]
```diff
@@ -75,23 +75,42 @@ public class Sync extends AbstractAirbyteConnection implements RunnableTask<Sync
...
+                            .create(
+                                    HttpMethod.POST,
+                                    UriTemplate
+                                            .of("/api/v1/connections/sync/")
+                                            .toString()
+                            )
+                            .body(Map.of("connectionId", runContext.render(this.connectionId))),
+                    Argument.of(JobInfo.class)
+            );
+        }catch(SyncAlreadyRunningException e) {
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/airbyte/connections/Sync.java:104]:
```suggestion
        } catch(SyncAlreadyRunningException e) {
```

File: src/main/java/io/kestra/plugin/airbyte/connections/Sync.java [hunk 4]
```diff
@@ -127,5 +148,10 @@ public static class Output implements io.kestra.core.models.tasks.Output {
             title = "The jobId created"
         )
         private final Long jobId;
+
+        @Schema(
+                title = "Whether a sync was already running or not"
+        )
+        private final Boolean alreadyRunning;
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/airbyte/connections/Sync.java:155]: I wonder if communicating that a sync is already running has some usage? Would it be enought to log a warning?

**Review by loicmathieu — CHANGES_REQUESTED** [review]

File: src/main/java/io/kestra/plugin/airbyte/AbstractAirbyteConnection.java (hunk 1 above)
Comment (aballiet) [reply src/main/java/io/kestra/plugin/airbyte/AbstractAirbyteConnection.java:88]: I don't know 😅

File: src/main/java/io/kestra/plugin/airbyte/connections/Sync.java (hunk 4 above)
Comment (aballiet) [reply src/main/java/io/kestra/plugin/airbyte/connections/Sync.java:155]: Yes it has, in my case I don't want to perform same execution flow if the sync is already running or not 🤔

File: src/main/java/io/kestra/plugin/airbyte/AbstractAirbyteConnection.java (hunk 1 above)
Comment (aballiet) [reply src/main/java/io/kestra/plugin/airbyte/AbstractAirbyteConnection.java:88]: Should I only test the Status Code ?

Comment (loicmathieu) [reply src/main/java/io/kestra/plugin/airbyte/AbstractAirbyteConnection.java:88]: If there is nothing clearly documented on theire side, better to keep it as is.

**Review by loicmathieu — APPROVED** [review]
LGTM

**Pull request merged**